jmxtrans-output/jmxtrans-output-kafka/src/main/java/com/googlecode/jmxtrans/model/output/kafka/KafkaWriter.java

```diff
@@ -92,10 +98,10 @@ public KafkaWriter(
 						DEFAULT_ROOT_PREFIX);
 		// Setting all the required Kafka Properties
 		Properties kafkaProperties =  new Properties();
-		kafkaProperties.setProperty("metadata.broker.list", Settings.getStringSetting(settings, "metadata.broker.list", null));
-		kafkaProperties.setProperty("zk.connect", Settings.getStringSetting(settings, "zk.connect", null));
-		kafkaProperties.setProperty("serializer.class", Settings.getStringSetting(settings, "serializer.class", null));
-		this.producer= new Producer<>(new ProducerConfig(kafkaProperties));
+		kafkaProperties.setProperty(BOOTSTRAP_SERVERS_CONFIG, Settings.getStringSetting(settings, BOOTSTRAP_SERVERS_CONFIG, null));
```

Comment:
Whatever the default settings may be, moving from the Scala producer to the Java producer breaks compatibility as settings are completely different.

About the default serializer I added, it makes sense because the record is of type <String,String>.

About moving from settings map to explicit parameters, IMHO it's not a good idea. At the moment, this class only exposes the mandatory settings, but there are many more which are useful (acks, retries, bulk size, SSL, SASL auth...). In the end, it would mean adding many fields.